### PR TITLE
[thread-cert] add case 8.1.6 using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**
Add provisionURL set and check
Check all JOIN_ENT & JOIN_FIN

### CODE REVIEW

File: tests/scripts/thread-cert/command.py [hunk 1]
```diff
@@ -643,28 +643,39 @@ def get_joiner_udp_port_in_discovery_response(command_msg):
...
+    assert join_fin_rsq.type == mesh_cop.MeshCopMessageType.JOIN_FIN_RSP
+    rsq_state = assert_contains_tlv(join_fin_rsq.tlvs, CheckType.CONTAIN, mesh_cop.State)
+    assert rsq_state.state == state
 
 
 def check_joiner_router_commissioning_messages(commissioning_messages):
     """Verify COAP messages sent by joiner router while commissioning process.
     """
-    assert any(msg.type == mesh_cop.MeshCopMessageType.JOIN_ENT_NTF for msg in commissioning_messages)
+    assert len(commissioning_messages) >= 4
```
Comment (librasungirl) [inline tests/scripts/thread-cert/command.py:676]: When joiner router is not collapsed with commissioner role, there might only be two logs (send JOIN_ENT.ntf, and recv joiN_ENT.rsp), the assertion might fail here?

File: tests/scripts/thread-cert/command.py [hunk 2]
```diff
@@ -643,28 +643,39 @@ def get_joiner_udp_port_in_discovery_response(command_msg):
...
+    join_ent_rsp = commissioning_messages[3]
     assert join_ent_rsp.type == mesh_cop.MeshCopMessageType.JOIN_ENT_RSP
 
 
-def check_commissioner_commissioning_messages(commissioning_messages):
+def check_commissioner_commissioning_messages(commissioning_messages, state=mesh_cop.MeshCopState.ACCEPT):
     """Verify COAP messages sent by commissioner while commissioning process.
     """
-    assert any(msg.type == mesh_cop.MeshCopMessageType.JOIN_FIN_RSP for msg in commissioning_messages)
+    assert len(commissioning_messages) >= 4
```
Comment (librasungirl) [inline tests/scripts/thread-cert/command.py:666]: when commissioner role is not collapsed with joiner router rule, there might be only two logs (recv JOIN_FIN.req and send JOIN_FIN.rsp), the assertion might fail here?

File: tests/scripts/thread-cert/command.py [hunk 3]
```diff
@@ -643,28 +643,39 @@ def get_joiner_udp_port_in_discovery_response(command_msg):
...
+    assert rsq_state.state == state
 
 
 def check_joiner_router_commissioning_messages(commissioning_messages):
     """Verify COAP messages sent by joiner router while commissioning process.
     """
-    assert any(msg.type == mesh_cop.MeshCopMessageType.JOIN_ENT_NTF for msg in commissioning_messages)
+    assert len(commissioning_messages) >= 4
+    join_fin_ntf = commissioning_messages[2]
+    assert join_fin_ntf.type == mesh_cop.MeshCopMessageType.JOIN_ENT_NTF
```
[the remark this answers is not in the document]
Comment (FiveDimensions) [reply tests/scripts/thread-cert/command.py]: improper variable name， modified

File: tests/scripts/thread-cert/command.py (hunk 1 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/command.py]: delete assert here

File: tests/scripts/thread-cert/command.py (hunk 2 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/command.py]: add message list len condition to judge

**Review by librasungirl — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Comment by jwhui** [comment]
@FiveDimensions , please help resolve conflicts.

**Comment by jwhui** [comment]
@FiveDimensions , the packet verification tests seem to be failing. Can you take a look?

**Comment by FiveDimensions** [comment]
@jwhui , sure, I am checking the packet files

**Comment by FiveDimensions** [comment]
@jwhui , due to choose the wrong thread_meshcop.tlv.steering_data type in conflict resolve, fixed now

**Review by jwhui — APPROVED** [review]

**Pull request merged**